doc/api/n-api.md

```diff
@@ -192,33 +192,62 @@ napi_value create_addon(napi_env env);
 ```C
 // addon.c
 #include "addon.h"
+
+#define NAPI_CALL(env, call)                             \
+  do {                                                   \
+    napi_status status = call;                           \
+    if (status != napi_ok) {                             \
+      const napi_extended_error_info* error_info = NULL; \
+      napi_get_last_error_info((env), &error_info);      \
+      bool is_pending;                                   \
+      napi_is_exception_pending((env), &is_pending);     \
+      if (!is_pending) {                                 \
+        const char* message = "empty_error_message";     \
+        if (error_info->error_message != NULL) {         \
+          message = error_info->error_message;           \
+        }                                                \
+        napi_throw_error((env), NULL, message);          \
+        return NULL;                                     \
+      }                                                  \
+    }                                                    \
+  } while(0)
+
+static napi_value
+DoSomethingUseful(napi_env env, napi_callback_info info) {
+  // Do something useful.
+  return NULL;
+}
+
 napi_value create_addon(napi_env env) {
   napi_value result;
-  assert(napi_create_object(env, &result) == napi_ok);
+  NAPI_CALL(env, napi_create_object(env, &result));
+
   napi_value exported_function;
-  assert(napi_create_function(env,
-                              "doSomethingUseful",
-                              NAPI_AUTO_LENGTH,
-                              DoSomethingUseful,
-                              NULL,
-                              &exported_function) == napi_ok);
-  assert(napi_set_named_property(env,
-                                 result,
-                                 "doSomethingUseful",
-                                 exported_function) == napi_ok);
+  NAPI_CALL(env, napi_create_function(env,
+                                      "doSomethingUseful",
+                                      NAPI_AUTO_LENGTH,
+                                      DoSomethingUseful,
+                                      NULL,
+                                      &exported_function));
+
+  NAPI_CALL(env, napi_set_named_property(env,
+                                         result,
+                                         "doSomethingUseful",
+                                         exported_function));
+
   return result;
 }
 ```
 
 ```C
 // addon_node.c
 #include <node_api.h>
+#include "addon.h"
 
-static napi_value Init(napi_env env, napi_value exports) {
+/* napi_value */
+NAPI_MODULE_INIT(/* napi_env env, napi_value exports */) {
```

Comment:
The comment here is also a bit confusing. It doesn't seem like `NAPI_MODULE_INIT()` expects arguments (although `env` is used on the next line).

Reply:
The intention here is to illustrate that the function body may behave as if the variables in comment were formal arguments. I'm not sure how better to illustrate this. I'll try adding an explanation in English instead.